tests/test_sft_trainer.py

```diff
@@ -423,6 +423,65 @@ def test_data_collator_completion_lm(self):
         result_text = self.tokenizer.decode(batch["input_ids"][0, last_pad_idx + 1 :])
         self.assertTrue(result_text == "I have not been masked correctly.")
 
+    def test_sft_trainer_infinite_with_model(self):
```

Comment:
What do you think about double checking here for both cases if we trained as many steps as we expect?